src/Workspaces/Remote/Core/Services/TemporaryWorkspace.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    /// <summary>
+    /// this let us have isolated workspace services between solutions such as option services
```

Comment:
option is stored in workspace. which mean if one API call changes options, all other API call gets the changed options.

temporary workspace make sure options are isolated between API calls.

I think it is current limitation of our in proc workspace. where if one changes option, it will affect all services while they are running. basically our data API such as intellisense provider, extract method service and etc are not actually stateless (same result with same input) since if somebody changes option in completely unrelated code path, it affects that API.

temporary workspace basically ensure that doesn't happen.